src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -364,9 +364,11 @@ public async Task SslStream_NegotiateClientCertificateAsync_ServerDontDrainClien
             using CancellationTokenSource cts = new CancellationTokenSource();
             cts.CancelAfter(TestConfiguration.PassingTestTimeout);
 
-            (SslStream client, SslStream server) = TestHelper.GetConnectedSslStreams();
-            using (client)
-            using (server)
+            (Stream clientStream, Stream serverStream) = TestHelper.GetConnectedStreams();
+
+            // use RandomReadWriteStream in the middle to receive incomplete TLS frames
```

Comment:
Why random rather than something deterministic?

Reply:
It may be difficult to get the half-frame consumed scenario with current setup. So I recommended this as option.

Reply:
Is there any chance this may sporadically produce a value that causes the test to fail?

Reply:
The old code would write one frame, read it and not consume completely. The new code will write two frames (as it is too big to fit to one). I think there is chance that The receiving part reads one and calls NegotiateClientCertificateAsync() before next one arrives. The windows is slim but probably exist. 

I was thinking about his more and it would be _nice_ to make it deterministic. (for this, upcoming work and well as improving test coverage for existing code) It also feels that it would be difficult for this single test to cover both conditions. I think we could make Stream that just buffers all write data and and we add Flush(int len) to write specific ambient of data and perhaps method to give you outstanding amount. 
With that, we should be able to cover both conditions deterministically.

Reply:
We chat with @rzikm offline and he will make the test deterministic. I think that will be handy for some other cases as well.